client/src/worker_registry/mod.rs

```diff
@@ -113,16 +113,32 @@ impl SlotManagerImpl {
 #[derive(Default, Debug)]
 pub struct SlotManager {
     manager: RwLock<SlotManagerImpl>,
+    /// True if this manager is associated with a mock client.
+    mock: bool,
```

Comment:
Rather than adding this here, I think it would be a little cleaner to add `is_mock` to the `WorkerClient` trait and simply not instantiate / merge the external WFT channel if it is.

Reply:
Moved to WorkerClient.

The only reason to merge the closed stream is to satisfy the Rust typing gods... Select<Stream>  is not the same type as Stream, is there a way to cast it into a vanilla Stream?

Reply:
Thanks for your great suggestion of using an Either stream! Channel not merged.